test/error/ErrorTest.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from "assert";
+import { MatterError } from "../../src/error/MatterError";
+import { StatusCode } from "../../src/matter/interaction/InteractionMessages";
+import { tryCatch, tryCatchAsync } from "../../src/error/TryCatchHandler";
+import { StatusResponseError } from "../../src/matter/interaction/InteractionErrors";
+
+class SubMatterError extends MatterError {}
+class SubSubMatterError extends SubMatterError {}
+class OtherMatterError extends MatterError {}
+
+describe("Errors", () => {
+
+    context("Test tryCatch method", () => {
+        it("tryCatch without error return value", () => {
+            const error = tryCatch(() => {
+                return "ok";
+            },
+            MatterError, "caught");
+
+            assert.equal(error, "ok");
+        });
+
+        it("tryCatch with expected error, uses fallback value", () => {
+            const error = tryCatch(() => {
+                throw new SubMatterError("test");
+            },
+            SubMatterError, "caught");
+
+            assert.equal(error, "caught");
+        });
+
+        it("tryCatch with unexpected error, throw error", () => {
+            try {
```

Comment:
It is better to unit test exception throwing with assert.throws:

```
const test = () => {
        const error = tryCatch(() => {
                    throw new Error("test");
                },
                SubMatterError, "caught");
};
assert.throws(test, new Error("test"));
```

Same in the rest of the class